gadgets/snapshot_file/README.mdx

```diff
@@ -0,0 +1,128 @@
+---
+title: snapshot_file
+sidebar_position: 0
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+The `snapshot_file` gadget shows open files for running processes.
+
+## Getting started
+
+Running the gadget:
+
+<Tabs groupId="env">
+    <TabItem value="kubectl-gadget" label="kubectl gadget">
+        ```bash
+        $ kubectl gadget run ghcr.io/inspektor-gadget/gadget/snapshot_file:%IG_TAG% [flags]
+        ```
+    </TabItem>
+
+    <TabItem value="ig" label="ig">
+        ```bash
+        $ sudo ig run ghcr.io/inspektor-gadget/gadget/snapshot_file:%IG_TAG% [flags]
+        ```
+    </TabItem>
+</Tabs>
+
+## Flags
+
+### `--paths`
+
+Show full file paths (by default, only dentry names are shown).
+
+Default value: `"false"`
+
+### `--file-type-mask`
+
+Filter which file types to display using a bitmask.  
+By default, only **regular files** and **directories** are shown.
+
+Supported file types:
+
+- Regular file
+- Directory
+- Pipe
+- Socket
+- BPF map
+- Character device
+- Block device
+- Symlink
+- Other
+
+## Guide
+
+Run a pod / container:
+
+<Tabs groupId="env">
+<TabItem value="kubectl-gadget" label="kubectl gadget">
+```bash
+$ kubectl run --restart=Never --image=busybox test-snapshot-file -- sh -c 'sleep inf'
+pod/test-snapshot-file created
+````
+
+</TabItem>
+
+<TabItem value="ig" label="ig">
+```bash
+$ docker run --name test-snapshot-file -d busybox /bin/sh -c 'sleep inf'
+...
+```
+</TabItem>
+</Tabs>
+
+Then, open a file in the container:
+
+<Tabs groupId="env">
+<TabItem value="kubectl-gadget" label="kubectl gadget">
+```bash
+$ kubectl exec -it test-snapshot-file -- sh -c 'echo hello > /tmp/demo.txt && sleep inf'
+```
+</TabItem>
+
+<TabItem value="ig" label="ig">
+```bash
+$ docker exec -it test-snapshot-file sh -c 'echo hello > /tmp/demo.txt && sleep inf'
+```
+</TabItem>
+</Tabs>
+
+Now, run the gadget and see how it shows open files:
+
+<Tabs groupId="env">
+<TabItem value="kubectl-gadget" label="kubectl gadget">
+```bash
+$ kubectl gadget run snapshot_file:%IG_TAG%
+K8S.NODE        K8S.NAMESPACE   K8S.PODNAME         K8S.CONTAINERNAME   COMM   PID   TID   FD   TYPE      INODE      PATH
+minikube-docker default         test-snapshot-file  test-snapshot-file  sh     5432  5432  1    REGULAR   1234567    demo.txt
+^C
+```
+</TabItem>
+
+<TabItem value="ig" label="ig">
+```bash
+$ sudo ig run snapshot_file:%IG_TAG% -c test-snapshot-file
+RUNTIME.CONTAINERNAME   COMM   PID   TID   FD   TYPE      INODE      PATH
+test-snapshot-file      sh     5321  5321  1    REGULAR   1234567    demo.txt
```

Comment:
The `MODE` and `FLAGS` fields are not present here in the output (Same for the K8s output)